requirements.txt

```diff
@@ -47,80 +47,74 @@ flask-login==0.6.3
     # via -r requirements.in
 flask-wtf==1.2.1
     # via -r requirements.in
-greenlet==2.0.2
-    # via
-    #   -r requirements.in
-    #   sqlalchemy
-idna==3.3
+greenlet==3.0.3
+    # via sqlalchemy
+idna==3.6
     # via requests
-importlib-metadata==6.3.0
+importlib-metadata==7.1.0
     # via
     #   flask
     #   markdown
 itsdangerous==2.1.2
     # via
     #   flask
     #   flask-wtf
-jinja2==3.1.2
+jinja2==3.1.3
     # via
     #   -r requirements.in
     #   flask
     #   flask-babel
-lxml==4.9.1
+lxml==5.2.1
     # via feedgen
-mako==1.2.2
+mako==1.3.2
     # via alembic
-markdown==3.5.1
+markdown==3.6
     # via -r requirements.in
-markupsafe==2.1.1
+markupsafe==2.1.5
     # via
     #   jinja2
     #   mako
     #   werkzeug
     #   wtforms
-mypy==0.971
+mypy==1.9.0
     # via sqlalchemy
-mypy-extensions==0.4.3
+mypy-extensions==1.0.0
     # via mypy
 nose==1.3.7
     # via pyutilib
-packaging==21.3
-    # via redis
```

Comment:
This package is still required by sphinx in form `packaging>=21.0`
Any new CKAN setup(inside test containers) will get the latest packaging version - v24. But all existing instances will continue to use previously pinned v21.3(unless you install dev-requirements with eager update strategy)

v21 of packaging had LegacyVersion class(it was removed in [v22](https://packaging.pypa.io/en/stable/changelog.html#id5)) and it will cause a type error because you removed `if not isinstance(version, Version):` check from app_globals. 

Let's add `packaging>=22` or `==24` to dev-requirements to be sure that no `LegacyVersion` appears in code

Reply:
Actually, we are using `packaging` in [app_globals](https://github.com/ckan/ckan/blob/a63839511e2d0cf93e3ff6743f00f694accb5f22/ckan/lib/app_globals.py#L9), [cli](https://github.com/ckan/ckan/blob/a63839511e2d0cf93e3ff6743f00f694accb5f22/ckan/cli/generate.py#L7) and the [toolkit](https://github.com/ckan/ckan/blob/a63839511e2d0cf93e3ff6743f00f694accb5f22/ckan/plugins/toolkit.py#L18) so it should be a first level requirement in requirements.in, right? Otherwise we'll rely on people installing dev-requirements.txt, which should not be necessary. I'll add it.